src/logicTemplates/limitorder.js

```diff
@@ -0,0 +1,113 @@
+        let injectionVector =  [maxFee, minTrade, assetid, ratd, ratn, expiryRound, owner];
+        let injectionTypes = [templates.valTypes.INT, templates.valTypes.INT, templates.valTypes.INT, templates.valTypes.INT, templates.valTypes.INT, templates.valTypes.INT, templates.valTypes.ADDRESS];
+        let injectedBytes = templates.inject(referenceProgramBytes, referenceOffsets, injectionVector, injectionTypes);
+        this.programBytes = injectedBytes;
+        let lsig = new logicSig.LogicSig(injectedBytes, undefined);
+        this.address = lsig.address();
+        this.owner = owner;
+        this.assetid = assetid;
+    }
+
+    /**
+     * returns the program bytes
+     * @returns {Uint8Array}
+     */
+    getProgram() {
+        return this.programBytes;
+    }
+
+    /**
+     * returns the string address of the contract
+     * @returns {string}
+     */
+    getAddress() {
+        return this.address;
+    }
+
+    /**
+     * returns a group transactions array which transfer funds according to the contract's ratio
+     * @param {int} assetAmount: the amount of assets to be sent
+     * @param {Uint8Array} contract: byteform of the contract from the payer
+     * @param {Uint8Array} secretKey: secret key for signing transaction
+     * @param {int} fee: the fee per byte to pay in microAlgos
+     * @param {int} algoAmount: number of algos to transfer
+     * @param {int} firstRound: the first round on which these txns will be valid
+     * @param {int} lastRound: the last round on which these txns will be valid
+     * @param {string} genesisHash: the b64-encoded genesis hash indicating the network for this transaction
+     * @returns {Uint8Array}
+     * the first payment sends money (Algos) from contract to the recipient (we'll call him Buyer), closing the rest of the account to Owner
+     * the second payment sends money (the asset) from Buyer to the Owner
+     * these transactions will be rejected if they do not meet the restrictions set by the contract
+     */
+    getSwapAssetsTransaction(assetAmount, contract, secretKey, fee, algoAmount, firstRound, lastRound,  genesisHash) {
+        let buyerKeyPair = nacl.keyPairFromSecretKey(secretKey);
+        let buyerAddr = address.encode(buyerKeyPair.publicKey);
+
+        let noCloseRemainder = undefined;
+        let noAssetRevocationTarget = undefined;
+        let tx1 = algosdk.makePaymentTxn(this.address, buyerAddr, fee, algoAmount, noCloseRemainder, firstRound, lastRound, undefined, genesisHash, undefined);
+        let tx2 = algosdk.makeAssetTransferTxn(buyerAddr, this.owner, noCloseRemainder, noAssetRevocationTarget, fee, assetAmount, firstRound, lastRound, undefined, genesisHash, undefined, this.assetid);
+
+        let txns = [tx1, tx2];
+        let txGroup = algosdk.assignGroupID(txns);
+
+        let logicSig = algosdk.makeLogicSig(this.getProgram(), undefined); // no args
```

Comment:
I think that this line does the wrong thing. Maybe you've intended to use the `contract`?